websockets/py35/server.py

```diff
@@ -0,0 +1,21 @@
+async def __aenter__(self):
+    return await self
+
+
+async def __aexit__(self, exc_type, exc_value, traceback):
+    self.ws_server.close()
+    await self.ws_server.wait_closed()
+
+
+async def __await_impl__(self):
+    # Duplicated with __iter__ because __await__ must call coroutines with
+    # await in Python 3.7 while __iter__ must call them with yield from to
+    # preserve compatibility with Python 3.4.
```

Comment:
One more thing: I'm not sure this comment (and others like it in this PR) explain the situation completely accurately. For example, it doesn't say anything about why 3.6 works but 3.7 doesn't. I think you want to be saying something about the return types, which changed from 3.6 to 3.7.

Reply:
Good point.

Thankfully all this is going away when Python 3.4 is EOL in about a year.

Reply:
I simplified the comments.